src/cpp/server/server.cc

```diff
@@ -287,12 +287,14 @@ void Server::Wait() {
 }
 
 void Server::PerformOpsOnCall(CallOpBuffer* buf, Call* call) {
-  static const size_t MAX_OPS = 8;
-  size_t nops = MAX_OPS;
-  grpc_op ops[MAX_OPS];
-  buf->FillOps(ops, &nops);
-  GPR_ASSERT(GRPC_CALL_OK ==
-             grpc_call_start_batch(call->call(), ops, nops, buf));
+  if (call->call()) {
```

Comment:
To be clear - I don't believe there's a stuckness issue in the case that you're trying to fix.
I do believe that this fix opens the way for a stuckness issue in the future, as we now silently skip doing something but make an implicit promise that an event will be delivered.
Is there a point higher in the call graph that this check could be made?
Is the code validly broken and we should just issue a better error message?

Happy to sit down and look at this once we're both at the office if you'd like.